featurebyte/service/feature_table_cache.py

```diff
@@ -726,13 +728,30 @@ async def create_view_from_cache(
                 )
             )
         )
-        create_view_exr = expressions.Create(
+        create_expr = expressions.Create(
             this=expressions.Table(
                 this=get_fully_qualified_table_name(output_view_details.dict()),
             ),
             kind="VIEW",
             expression=select_expr,
             replace=False,
         )
-        sql = sql_to_string(create_view_exr, source_type=db_session.source_type)
-        await db_session.execute_query(sql)
+        sql = sql_to_string(create_expr, source_type=db_session.source_type)
+        try:
+            await db_session.execute_query_long_running(sql)
+        except:  # pylint: disable=bare-except
+            logger.info(
+                "Failed to create view. Trying to create a table instead",
+                extra={"observation_table_id": observation_table.id},
+                exc_info=True,
+            )
+            create_expr = expressions.Create(
+                this=expressions.Table(
+                    this=get_fully_qualified_table_name(output_view_details.dict()),
+                ),
+                kind="TABLE",
+                expression=select_expr,
+                replace=False,
+            )
+            sql = sql_to_string(create_expr, source_type=db_session.source_type)
+            await db_session.execute_query_long_running(sql)
```

Comment:
Curious whether we need to clean this up?

Reply:
We need to clean this up when the feature table is deleted. That is a good point, we need to make sure the `is_view` field for materialised view is tracked properly. I will also update session drop table to handle both table and view automatically.